src/main/java/com/amazon/ion/impl/IonReaderContinuableCoreBinary.java

```diff
@@ -689,12 +689,18 @@ public double doubleValue() {
                 // Note: there is no need to check for other lengths here; the type ID byte is validated during next().
                 value = bytes.getDouble();
             }
-        }  else if (valueTid.type == IonType.DECIMAL) {
+        } else if (valueTid.type == IonType.DECIMAL) {
             scalarConverter.addValue(decimalValue());
             scalarConverter.setAuthoritativeType(_Private_ScalarConversions.AS_TYPE.decimal_value);
             scalarConverter.cast(scalarConverter.get_conversion_fnid(_Private_ScalarConversions.AS_TYPE.double_value));
             value = scalarConverter.getDouble();
             scalarConverter.clear();
+        } else if (valueTid.type == IonType.INT) {
+            scalarConverter.addValue(longValue());
```

Comment:
Do we know that the previous implementation only worked for Ion Ints that fit in a long?

Reply:
I double-checked and unfortunately it looks like it would work, so I added support.